extensions/WSLExtension/Contracts/IWslManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using WSLExtension.DistributionDefinitions;
+using WSLExtension.Models;
+
+namespace WSLExtension.Contracts;
+
+/// <summary>
+/// Used to interact between the WSL dev environment compute systems and the WSL mediator.
+/// </summary>
+public interface IWslManager
+{
+    public event EventHandler<HashSet<string>>? DistributionStateSyncEventHandler;
+
+    /// <summary> Gets a list of all registered WSL distributions on the machine.</summary>
+    public Task<List<WslComputeSystem>> GetAllRegisteredDistributionsAsync();
+
+    /// <summary>
+    /// Gets a list of objects that each contain metadata about a wsl distribution
+    /// that is not currently registered on the machine and is available to install.
+    /// </summary>
+    public Task<List<DistributionDefinition>> GetAllDistributionsAvailableToInstallAsync();

```

Comment:
If I made a method called `GetDistributionsAvalibleToInstallAsync` note the missing `All` would the method do anything different than this one?

I'm wondering why say `All` Isn't `All` implied in this case?

Reply:
I guess in this case I was just being explicit about what the method does, but I think I'll leave it as is since it still gets the point across without too much cognitive load on the reader.